### DISCUSSION TITLE
Conditional forward question

### DISCUSSION THREAD

**Discussion by WeihanLi** [TechnitiumSoftware/DnsServer#827] [Q&A] [answered]
Thanks very much for the amazing work.

I'm trying to setup my DNS server for some domain only, for other domains forward to the upstream DNS server

I added a conditional forwarder zone

![](https://github.com/TechnitiumSoftware/DnsServer/assets/7604648/e6f820bb-e1a8-4f11-a931-a285e8bc0b2f)

and forward settings configured as follows

![](https://github.com/TechnitiumSoftware/DnsServer/assets/7604648/54f92abd-9961-46db-9a21-c80790fdaef4)

while it seemed it could resolve `weihanli.top` domain only, for other domains would failed

![](https://github.com/TechnitiumSoftware/DnsServer/assets/7604648/f4868cd8-782b-4aba-a074-ed83db080474)

Is there any config I missed? How should I configure if I want these work both

**Comment by ShreyasZare** [comment, marked as answer]
Thanks for the post. It seems that you have disabled recursion from the Settings > Recursion section. You need to allow it for the DNS server to be able to respond to domain names that are not locally hosted.

**Reply by WeihanLi** [reply]
Oh yeah, that's it, works now, thanks very much

**Reply by ShreyasZare** [reply]
Good to know you got it working.
